File: YoutubeLinks.Shared/Features/Links/Commands/DownloadSingleLink.cs
using FluentValidation;
using MediatR;
using Microsoft.Extensions.Localization;
using YoutubeLinks.Shared.Features.Links.Helpers;
using YoutubeLinks.Shared.Localization;

namespace YoutubeLinks.Shared.Features.Links.Commands
{
    public static class DownloadSingleLink
    {
        public class Command : IRequest<YoutubeFile>
        {
            public string Url { get; set; }
            public YoutubeFileType YoutubeFileType { get; set; }
        }

        public class Validator : AbstractValidator<Command>
        {
            public Validator(IStringLocalizer<ValidationMessage> localizer)
            {
                RuleFor(x => x.Url)
                    .NotEmpty()
                    .WithMessage(x => localizer[nameof(ValidationMessageString.UrlNotEmpty)])
                    .Matches(ValidationConsts.YoutubeVideoUrlRegex)
                    .WithMessage(x => localizer[nameof(ValidationMessageString.VideoUrlMatchesRegex)]);

                RuleFor(x => x.YoutubeFileType)
                    .IsInEnum()
                    .WithMessage(x => localizer[nameof(ValidationMessageString.YoutubeFileTypeIsInEnum)]);
            }
        }
    }
}

File: youtube-links/src/shared/features/links/commands/DownloadSingleLink.ts
import { ValidationConsts } from '@/shared/localization/ValidationConsts';
import { YoutubeFileType, isValidYoutubeFileTypeValue } from '../helpers/YoutubeFileType';

export namespace DownloadSingleLink {
  export interface Command {
    url: string;
    youtubeFileType: YoutubeFileType;
  }

  export const Validation = {
    url: {
      notEmpty: (v: string) => !!v || 'Youtube video url should not be empty.',
      matchesYoutubeVideoUrlRegex: (v: string) =>
        ValidationConsts.matchesYoutubeVideoUrlRegex(v) ||
        'This is not a valid link to the YouTube video.',
    },
    youtubeFileType: {
      isInEnum: (v: string) =>
        isValidYoutubeFileTypeValue(v) ||
        `YoutubeFileType has a range of values which does not include: ${v}.`,
    },
  };
}
